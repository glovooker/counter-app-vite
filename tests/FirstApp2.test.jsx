import { render, screen } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';

describe('Tests on <FirstApp />', () => {
  const title = 'Hola, Soy Goku';
  const subTitle = 'Soy un subtÃ­tulo';

  test('Must match snapshot', () => {
    const { container } = render(<FirstApp title={title} />);

    expect(container).toMatchSnapshot();
  });

  test('Must show title "Hola, Soy Goku"', () => {
    render(<FirstApp title={title} />);
    expect(screen.getByText(title)).toBeTruthy();
    // screen.debug();
  });

  test('Must show title in h1', () => {
    render(<FirstApp title={title} />);

    expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(
      title
    );
  });

  test('Must show subtitle sent through props', () => {
    render(
      <FirstApp
        title={title}
        subTitle={subTitle}
      />
    );

    expect(screen.getAllByText(subTitle).length).toBe(2);
  });
});
