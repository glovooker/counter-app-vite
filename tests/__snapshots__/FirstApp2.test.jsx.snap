// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <FirstApp /> Must match snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, Soy Goku
  </h1>
  <p>
    No hay subtítulo
  </p>
  <p>
    No hay subtítulo
  </p>
  <p>
    Gabriel Lobo
  </p>
</div>
`;
